### ISSUE TITLE
Struggling with MySQL "Text" (Blob) in the schema

### ISSUE THREAD

**Issue by oderwat** [tathougies/beam-mysql#8]
Hi. I am very new to Haskell and also new to Beam, so it may be just my ignorance but I can't get a MySQL TEXT field to work. 

The "Text" type in the schema throws a ParserError exception. Basically saying SQLType "Blob" and HaskellType "Text" are incompatible. I guess because the Blob type MySQL uses is actually more a (Binary/)ByteString. So I tried to use ByteString in the schema but then I get:

```
No instance for (Database.Beam.Backend.SQL.SQL92.HasSqlValueSyntax
                         MysqlValueSyntax Data.ByteString.Internal.ByteString)
```

What do I need to make "Text" / "Blob" fields with MySQL work?

**Comment by oderwat** [comment]
When I use `import qualified Data.ByteString.Char8 as SB` and `SB.ByteString` in the schema I can query the database with good results but Inserting (maybe I do it wrong) fails with: `No instance for (Database.Beam.Backend.SQL.SQL92.HasSqlValueSyntax
                         MysqlValueSyntax ByteString)`

**Comment by tathougies** [comment]
It looks like there's a missing instance. Give me a day or two, and I'll update it. 

Thanks for the report

**Comment by oderwat** [comment]
Hi @tathougies I am considering beam for a larger project which shares code between mysql and postgresql databases and would really like to check out beam in that respect. But without support for text datatypes this seems like a lost cause. Is the mysql implementation meant to be supported or should we move aways from it for anything more than a toy project?

**Comment by tathougies** [comment]
Hi @oderwat ... that's great. The mysql implementation is supported, but I mean, this is my personal project, and there's no company backing this. I'm happy to provide support on a voluntary basis, but if you *need* something implemented with time deadlines, we'll have to talk about contracting or bounties or something.

I am happy to commit any PR implementing the functionality you want. The solution is exactly as GHC says: to implement HasSqlValueSyntax for ByteString. I will do this as soon as I can.

**Comment by oderwat** [comment]
@tathougies Haha... I hear you. As the author of a quite popular VSC extension (indent-rainbow) I know about peoples demand and willingness to contribute (time or money).

We are more than willing to contribute / donate some money when we are going to use beam in projects. In fact I just added us as first (it seems) contributor on you Liberapay site. BTW: I strongly believe in supporting people in what they do if it helps me doing what I do or like :)

I probably could also create a PR but I am not sure about the "Text = Blob" stuff with MySQL and need to check that out. I think that a MySQL Text type probably should be DATA.TEXT and MySQL Blob should be DATA.BYTESTRING. But I am too new to understand the consequences and also had the impression that it is not possible (or hard) to distinguish between the TEXT and BLOB because of MySQL reporting the columns always as BLOB. I also saw other things which seemed not ideal for what I plan to do.
